src/proof/InterpolationContext.cc

```diff
@@ -9,14 +9,832 @@
+    std::vector<Lit> const & cl = n.getClause();
+    vec<PTRef> restricted_clause;
+
+    for (Lit l : cl) {
+        Var v = var(l);
+        icolor_t var_class = getVarClass(v);
+        assert(var_class == icolor_t::I_B || var_class == icolor_t::I_A || var_class == icolor_t::I_AB);
+        icolor_t var_color = var_class == icolor_t::I_B || var_class == icolor_t::I_A ? var_class
+                                                                                      : getSharedVarColorInNode(v, n);
+        if (var_color == icolor_t::I_A || var_color == icolor_t::I_AB) {
+            PTRef ptaux = varToPTRef(v);
+            if (sign(l)) {
+                ptaux = logic.mkNot(ptaux);
+            }
+            restricted_clause.push(ptaux);
+        }
+    }
+
+    PTRef cl_ptref = logic.mkNot(logic.mkOr(std::move(restricted_clause)));
+    PTRef implicant = logic.mkAnd({pmanager.getPartition(A_mask, PartitionManager::part::A), cl_ptref});
+
+    bool res = VerificationUtils(config, logic).impliesExternal(implicant, getPartialInterpolant(n));
+    assert(res);
+    return res;
+}
+
+bool SingleInterpolationComputationContext::verifyPartialInterpolantB(ProofNode const & n) {
+    // Check B /\ ~(C|b,ab) -> ~I, i.e., B /\ ~(C|b,ab) /\ I unsat
+    std::vector<Lit> const & cl = n.getClause();
+    vec<PTRef> restricted_clause;
+
+    for (Lit l : cl) {
+        Var v = var(l);
+        icolor_t var_class = getVarClass(v);
+        assert(var_class == icolor_t::I_B || var_class == icolor_t::I_A || var_class == icolor_t::I_AB);
+        icolor_t var_color = var_class == icolor_t::I_B || var_class == icolor_t::I_A ? var_class
+                                                                                      : getSharedVarColorInNode(v, n);
+        if (var_color == icolor_t::I_B || var_color == icolor_t::I_AB) {
+            PTRef ptaux = varToPTRef(v);
+            if (sign(l)) {
+                ptaux = logic.mkNot(ptaux);
+            }
+            restricted_clause.push(ptaux);
+        }
+    }
+
+    PTRef cl_ptref = logic.mkNot(logic.mkOr(std::move(restricted_clause)));
+    PTRef implicant = logic.mkAnd(pmanager.getPartition(A_mask, PartitionManager::part::B), cl_ptref);
+
+    bool res = VerificationUtils(config, logic).impliesExternal(implicant, logic.mkNot(getPartialInterpolant(n)));
+    assert(res);
+    return res;
+}
+
+
+SingleInterpolationComputationContext::SingleInterpolationComputationContext(
+        const SMTConfig & config,
+        Theory & theory,
+        TermMapper & termMapper,
+        PartitionManager & pmanager,
+        const ProofGraph & proof,
+        const ipartitions_t & A_mask
+) : logic(theory.getLogic()), config(config), pmanager(pmanager), proofGraph(proof), thandler(new THandler(theory, termMapper)), A_mask(A_mask) {
+    auto const & vars = proof.getVariables();
+    std::size_t varCounts = (*std::max_element(vars.begin(), vars.end())) + 1;
+    nodeData.clear();
+    AB_vars_mapping.clear();
```

Comment:
These calls to `clear` are not necessary. They are relicts from older version of the code where this was not a constructor but explicit `initialize` method.

Reply:
Should they be removed?